File: bench/Reference/file_parallel_read_large.java
import java.nio.file.*;
import java.util.concurrent.*;
import java.util.stream.*;
public class file_parallel_read_large {
    public static void main(String[] args) throws Exception {
        try (var ex = Executors.newVirtualThreadPerTaskExecutor()) {
            var futures = IntStream.rangeClosed(1, 4)
                .mapToObj(i -> ex.submit(() -> (long) Files.readString(Path.of("bench/data/chunk_" + i + ".txt")).length()))
                .toList();
            long total = 0;
            for (var f : futures) total += f.get();
            System.out.println(total);
        }
    }
}

File: bench/Reference/file_parallel_read_large.py
from concurrent.futures import ThreadPoolExecutor


def read_len(path):
    with open(path) as f:
        return len(f.read())


paths = [f"bench/data/chunk_{i}.txt" for i in range(1, 5)]
with ThreadPoolExecutor(4) as ex:
    print(sum(ex.map(read_len, paths)))
